pkg/controller/certificaterequests/venafi/venafi.go

```diff
@@ -66,8 +66,12 @@ func NewVenafi(ctx *controllerpkg.Context) *Venafi {
 func (v *Venafi) Sign(ctx context.Context, cr *cmapi.CertificateRequest, issuerObj cmapi.GenericIssuer) (*issuerpkg.IssueResponse, error) {
 	log := logf.FromContext(ctx, "sign")
 	log = logf.WithRelatedResource(log, issuerObj)
+	ns := issuerObj.GetObjectMeta().Namespace
+	if ns == "" {
+		ns = v.issuerOptions.ClusterResourceNamespace
+	}
 
-	client, err := v.clientBuilder(cr.Namespace, v.secretsLister, issuerObj)
+	client, err := v.clientBuilder(ns, v.secretsLister, issuerObj)
```

Comment:
This looks good to me, although instead of manually writing the 'switching' logic above you should be able to call `v.issuerOptions.ResourceNamespace(issuerObj)` and use that 😄 

Thanks for digging this out! 😅 

/ok-to-test